### PR TITLE
Update Storybook and pin Nuxt Storybook module version

### LINKED ISSUE #4963: Peer dependency issues with storybook
## Current Situation

There are many peer dependency issues with Storybook on `main`. Check this output from CI:

```
playwright-1  |  WARN  Issues with peer dependencies found
playwright-1  | .
playwright-1  | ├─┬ @nuxtjs/storybook 8.3.0-28771560.b2b9b16
playwright-1  | │ └─┬ @storybook-vue/nuxt 8.3.0-28771560.b2b9b16
playwright-1  | │   └─┬ @nuxt/vite-builder 3.13.2
playwright-1  | │     └─┬ vite-plugin-checker 0.8.0
playwright-1  | │       └── ✕ unmet peer vue-tsc@~2.1.6: found 2.1.4
playwright-1  | └─┬ @storybook-vue/nuxt 8.3.0-28771560.b2b9b16
playwright-1  |   ├─┬ @storybook/builder-vite 8.3.2
playwright-1  |   │ ├── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   │ └─┬ @storybook/csf-plugin 8.3.2
playwright-1  |   │   └── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   ├─┬ @storybook/vue3 8.3.2
playwright-1  |   │ ├── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   │ ├─┬ @storybook/components 8.3.2
playwright-1  |   │ │ └── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   │ ├─┬ @storybook/manager-api 8.3.2
playwright-1  |   │ │ └── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   │ ├─┬ @storybook/preview-api 8.3.2
playwright-1  |   │ │ └── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   │ └─┬ @storybook/theming 8.3.2
playwright-1  |   │   └── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  |   └─┬ @storybook/vue3-vite 8.3.2
playwright-1  |     └── ✕ unmet peer storybook@^8.3.2: found 8.2.9
playwright-1  | 
```

## Suggested Improvement

Resolve the peer dependency issues. Refer to the PR that closed #4733 for examples of how to investigate and fix this.

## Benefit

Reliable and correct dependency tree.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #4963 by @sarayourfriend

## Description

Now that the Nuxt Storybook module has released a stable release that works with Openverse, we can pin its version, and update the module's dependencies to match it.

TLDR; Updating the node version and removing `node_modules` volume from playwright docker-compose definition fixed the Storybook build problems in the CI, without making the tests run longer.

### More detailed description of the build problems

Unfortunately, the changes in Storybook v8.3.0-alpha.4 made it so that a wrong version of `ufo` file is being required (`.mjs` instead of `.cjs`), probably by the `nuxt-link.js` in storybook nuxt module, and this throws an `unexpected identifier export` error. I couldn't find a way of fixing this as there seems to be layers upon layers of rewriting/hijacking `require`. ~~The best solution I found here for now was to pin Storybook at v8.3.0-alpha.3.~~

When I tried to debug the issues, I ran `storybook build` locally, without `ov`. This time, I  didn't see the build issue, and noticed that my local version of Node is different than what we have in `package.json` engines. This is why I updated the Node version, and updated the Storybook versions to the latest versions here.
It still seems to be failing if we keep using the local pnpm folder as a volume. When I removed it in this PR, the Storybook build in CI was fixed.

This PR also sets the default port for Storybook in dev mode to match that of the production mode.

## Testing Instructions

The Storybook tests in the CI should pass.
Run `ov just frontend/run dev`. This should run the Storybook in dev mode at `http://localhost:54000`.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by dhruvkb — APPROVED** [review]
Nice to be back on a stable version instead of nightly.
